website/static/js/newAndNoteworthyPlugin.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * New and Noteworthy Projects
+ */
+
+var $ = require('jquery');
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+
+// CSS
+require('css/quick-project-search-plugin.css');
+
+// XHR config for apiserver connection
+var xhrconfig = function(xhr) {
+    xhr.withCredentials = true;
+};
+
+
+var newAndNoteworthy = {
+    controller: function() {
+        var self = this;
+        self.newAndNoteworthyNodes = m.prop([]);
+        self.popularNodes = m.prop([]);
+        self.contributorsMapping = {};
+
+        // Load new and noteworthy nodes
+        var newAndNoteworthyUrl = $osf.apiV2Url('nodes/' + window.contextVars.newAndNoteworthy + '/node_links/', {});
+        var newAndNoteworthyPromise = m.request({method: 'GET', url: newAndNoteworthyUrl, config: xhrconfig});
+        newAndNoteworthyPromise.then(function(result){
+            var numPopular = Math.min(result.data.length - 1, 4);
+            for (var l=0; l <= numPopular; l++) {
+                self.newAndNoteworthyNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+                  }
+            return newAndNoteworthyPromise;
+        });
+
+        // Load popular nodes
+        var popularUrl = $osf.apiV2Url('nodes/' + window.contextVars.popular + '/node_links/', {});
+        var popularPromise = m.request({method: 'GET', url: popularUrl, config: xhrconfig});
+        popularPromise.then(function(result){
+            var numPopular = Math.min(result.data.length - 1, 4);
+            for (var l=0; l <= numPopular; l++) {
+                self.popularNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+                  }
+            return popularPromise;
+        });
+
+        // Additional API call to fetch node link contributors
+        self.fetchContributors = function(nodeLink) {
+            var url = nodeLink.embeds.target_node.data.relationships.contributors.links.related.href;
+            var promise = m.request({method: 'GET', url : url, config: xhrconfig});
+            promise.then(function(result){
+                var contribNames = [];
+                result.data.forEach(function (contrib){
+                    contribNames.push(contrib.embeds.users.data.attributes.family_name);
+                });
+                var numContrib = result.links.meta.total;
+                var nodeId = nodeLink.id;
+                self.contributorsMapping[nodeId] = [contribNames, numContrib];
+            });
+        };
+
+        // Gets contrib family name for display
+        self.getFamilyName = function(i, node) {
+            return self.contributorsMapping[node.id][0][i];
+        };
+
+        // Returns name if one contrib, or adds et al if > 1
+        self.contribNameFormat = function(node, number) {
```

Comment:
It is similar but this function calls getFamilyName and the way family_names are retrieved in Quick Search Projects versus New and Noteworthy Projects are different.    For QuickSearchProjects, the first ten contributors names are going to be embedded in the node response so I can just grab them.  For New And Noteworthy Projects, the request retrieves the node links, and target nodes are embedded in the node links, but not contributors because we're not doing multiple levels of nesting.  Contributors are fetched on a separate request and put in a dictionary.